### ISSUE TITLE
Disk Encryption

### ISSUE THREAD

**Issue by heywoodlh** [AsahiLinux/docs#38]
LUKS disk encryption would be a nice-to-have option with the installer. However, I can understand this potentially not being an option as the installer has to play nicely with the existing APFS filesystem.

Perhaps [systemd-homed](https://systemd.io/HOME_DIRECTORY/) using [LUKS as the storage mechanism](https://wiki.archlinux.org/title/Systemd-homed) could be considered as a compromise as the user's home directory (perhaps the most relevant part of the filesystem to keep private) would be LUKS encrypted. It wouldn't be full-disk encryption, but still could get some of the way there for user's hoping to have an additional layer of security. However, I could understand this being undesired as it can introduce some complexity and unforeseen complications.

P.S. thanks for all the work that's gone to Asahi. It's so well thought out and I appreciate how polished it is despite being in Alpha. I can't wait for the finished product.

**Comment by b-crumb** [comment]
I would like to add on to this that I have for now written a relatively OK bunch of notes which should be sufficient to explain to anyone how to set up LUKS... this could be added to the wiki.

https://github.com/b-crumb/asahi-luks-notes/blob/main/ASAHILUKS.md

**Comment by davidalger** [comment]
For those using an Asahi Fedora remix, I posted a how-to on my blog covering the entire process of install, LUKS encryption, configure grub, rebuild initramfs, etc.

https://davidalger.com/posts/asahi-fedora-workstation-on-apple-silicon-with-luks-encrypted-root/

**TL:DR;** The LUKS encryption part itself is done in-place while booted from a Fedora USB drive allowing the Fedora root partition to remain unmounted:

1. Shrink the `btrfs` filesystem by 32 MiB
2. Encrypt in-place using `cryptsetup`

       cryptsetup reencrypt --encrypt --reduce-device-size 32M /dev/nvme0n1pX

4. Update grub config adding `rd.luks.uuid=LUKS_UUID` to `GRUB_CMDLINE_LINUX` and rebuild initramfs using dracut in a chroot.

**Comment by cortadocodes** [comment]
I'd love to switch to Asahi Linux but full-disk encryption is a basic security requirement for many workplaces. MacOS, Windows, and most phones have this available as standard. I love the work you've done but I'm not sure I can justify switching if this isn't implemented. Is there a timeframe on https://github.com/AsahiLinux/asahi-installer/pull/240 and is that full-disk or just home directory?

**Comment by MarcusWentz** [comment]
Updates on this? I know several people not trying out Asahi Linux yet since this critical security feature isn't added yet.

**Comment by tombh** [comment]
I wonder if we should change the title? Disk encryption is and always has been supported in Asahi. The issue is that it has to be set up manually, which whilst requiring some technical knowledge isn't any harder than say how you would do it in Arch Linux. Here are some guides:
* https://davidalger.com/posts/fedora-asahi-remix-on-apple-silicon-with-luks-encryption/
* https://github.com/b-crumb/asahi-notes/blob/main/December%202022/notes.md#LUKS

So what about if the title was changed to: "Installer option to automatically set up disk encryption"?

**Comment by cortadocodes** [comment]
@tombh do those result in full disk encryption rather than just home directory encryption? If so, I think the title should be "Installer option to automatically set up full disk encryption"

**Comment by tombh** [comment]
Yes. The guides use LUKS, which is kernel-level full disk encryption. And I am writing to you now from a LUKS-encrypted Asahi install.

**Comment by vedmant** [comment]
That's probably the main reason why I can't use Asahi as alternative to main OS secure environment. Manual LUKS setup is pretty complicated, requires booting from a separate USB drive, which I need to buy just to setup LUKS. And maybe an hour or two of my time to do this all.
